execute/plugin_usdc_test.go

```diff
@@ -0,0 +1,515 @@
+
+	"github.com/smartcontractkit/libocr/commontypes"
+	"github.com/smartcontractkit/libocr/offchainreporting2plus/ocr3types"
+	libocrtypes "github.com/smartcontractkit/libocr/ragep2p/types"
+
+	commonconfig "github.com/smartcontractkit/chainlink-common/pkg/config"
+	"github.com/smartcontractkit/chainlink-common/pkg/logger"
+	"github.com/smartcontractkit/chainlink-common/pkg/types"
+	cciptypes "github.com/smartcontractkit/chainlink-common/pkg/types/ccipocr3"
+	"github.com/smartcontractkit/chainlink-common/pkg/types/query/primitives"
+
+	"github.com/smartcontractkit/chainlink-ccip/chainconfig"
+	"github.com/smartcontractkit/chainlink-ccip/execute"
+	"github.com/smartcontractkit/chainlink-ccip/execute/exectypes"
+	"github.com/smartcontractkit/chainlink-ccip/execute/internal/gas/evm"
+	"github.com/smartcontractkit/chainlink-ccip/execute/report"
+	"github.com/smartcontractkit/chainlink-ccip/execute/tokendata"
+	"github.com/smartcontractkit/chainlink-ccip/internal/libs/slicelib"
+	"github.com/smartcontractkit/chainlink-ccip/internal/libs/testhelpers"
+	"github.com/smartcontractkit/chainlink-ccip/internal/mocks"
+	"github.com/smartcontractkit/chainlink-ccip/internal/mocks/inmem"
+	"github.com/smartcontractkit/chainlink-ccip/internal/plugintypes"
+	"github.com/smartcontractkit/chainlink-ccip/internal/reader"
+	mock_types "github.com/smartcontractkit/chainlink-ccip/mocks/execute/exectypes"
+	readermock "github.com/smartcontractkit/chainlink-ccip/mocks/pkg/contractreader"
+	"github.com/smartcontractkit/chainlink-ccip/pkg/consts"
+	"github.com/smartcontractkit/chainlink-ccip/pkg/contractreader"
+	readerpkg "github.com/smartcontractkit/chainlink-ccip/pkg/reader"
+	"github.com/smartcontractkit/chainlink-ccip/pluginconfig"
+	plugintypes2 "github.com/smartcontractkit/chainlink-ccip/plugintypes"
+)
+
+const (
+	address = "0x888888cfaebbed5554c3f36bfbd233f822e9455f"
+)
+
+func Test_USDC_Transfer(t *testing.T) {
+	ctx := context.Background()
+	lggr := logger.Test(t)
+
+	sourceChain := cciptypes.ChainSelector(sel.ETHEREUM_TESTNET_SEPOLIA.Selector)
+	destChain := cciptypes.ChainSelector(sel.ETHEREUM_MAINNET_BASE_1.Selector)
+
+	nodesSetup, server := setupSimpleTest(ctx, t, lggr, sourceChain, destChain)
+	defer server.Close()
+
+	nodes := make([]ocr3types.ReportingPlugin[[]byte], 0, len(nodesSetup))
+	for _, n := range nodesSetup {
+		nodes = append(nodes, n.node)
+	}
+
+	nodeIDs := make([]commontypes.OracleID, 0, len(nodesSetup))
+	for _, n := range nodesSetup {
+		nodeIDs = append(nodeIDs, n.node.ReportingCfg().OracleID)
+	}
+
+	runner := testhelpers.NewOCR3Runner(nodes, nodeIDs, nil)
+
+	// Contract Discovery round.
+	res, err := runner.RunRound(ctx)
+	require.NoError(t, err)
+	outcome, err := exectypes.DecodeOutcome(res.Outcome)
+	require.NoError(t, err)
+	require.Equal(t, exectypes.Initialized, outcome.State)
+
+	// Round 1 - Get Commit Reports
+	// One pending commit report only.
+	// Two of the messages are executed which should be indicated in the Outcome.
```

Comment:
This naming is kinda confusing to me - messages are executed, but they're not, right, since the report only gets generated in the `Filter` state and transmitted on-chain thereafter?

Reply:
Or are we saying they're pre-executed as part of the test setup?

Reply:
Seems like they're pre-executed, but wondering whats the significance of that for this test. Or is it just how the other tests are set up as well?

Reply:
This is how it's setup in other places